experimental/dds/tree2/src/feature-libraries/simple-tree/types.ts

```diff
@@ -331,25 +343,28 @@ export type ProxyNodeUnion<
  * Given a node's schema, return the corresponding object in the proxy-based API.
  * @alpha
  */
-export type ProxyNode<
+export type TypedNode<
```

Comment:
Can we change `TreeField` to `TypedField` so that it matches this one? Likewise with `TreeFieldInner` and `TreeNodeUnion`.

Or, they could all be `TypedTree*`.

Reply:
I'd rather just have TreeNode and TreeField, and not have two separate types for the generic and non generic ones. Note that TreeField already follows this pattern, so its name is well aligned with TreeNode in the 0 generic argument case.

Reply:
I think I see what you're going for. The way I see it, `TreeField`, `TreeNodeUnion` and `TypedTree` are in the same box (and should have the same prefix) because they optionally take a schema and may unbox leaf nodes. 

`TreeNode` does not do either of those things. I think it should have the outlier name, not `TypedTree`.